src/server/node_manager.c

```diff
@@ -7555,7 +7560,35 @@ free_nodes(job *pjob)
 		for (j=0; j<psvrmom->msr_jbinxsz; j++) {
 			if (psvrmom->msr_jobindx[j] == pjob) {
 				psvrmom->msr_jobindx[j] = NULL;
+				if (is_called_by_job_purge)
+					special_case = 1;
+			}
+		}
+
+		if (special_case) {
```

Comment:
Can we use is_called_by_job_purge directly? Maybe we can do away with this variable?